### PR TITLE
Experimental code for finding cache associativity

### PR DESCRIPTION
**Pull request by negara**

### CODE REVIEW

File: experimental/cache_associativity.cc [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#include <time.h>
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+#include <random>
+#include <string>
+
+// NOTE: the output does not make sense for now
+// TODO: give compilation instructions instead of including nonlocal files
```
Comment (sivachandra) [inline experimental/cache_associativity.cc:11]: As discussed offline, can you please add a CMake target to build this file?

**Review by sivachandra — COMMENTED** [review]
Mostly LGTM. Just a request to add a CMake target.

File: experimental/cache_associativity.cc (hunk 1 above)
Comment (negara) [reply experimental/cache_associativity.cc:11]: Added as a new demo to the CMakeLists.txt

**Review by sivachandra — APPROVED** [review]
On the CMake file, it says, "no newline at the end of file". Please fix.

**Pull request merged**